pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -0,0 +1,135 @@
+)
+
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+type ScopeType string
+
+const (
+	ClusterScopeType  ScopeType = "CLUSTER"
+	DataLakeScopeType ScopeType = "DATA_LAKE"
+)
+
+// AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+type AtlasDatabaseUserSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	Project ResourceRef `json:"projectRef"`
+
+	// DatabaseName is a Database against which Atlas authenticates the user. Must be equal to 'admin'
+	// as only SCRAM authentication is supported so far.
+	DatabaseName string `json:"databaseName"`
+
+	// DeleteAfterDate is a timestamp in ISO 8601 date and time format in UTC after which Atlas deletes the user.
+	// The specified date must be in the future and within one week of the time you make the API request.
+	DeleteAfterDate string `json:"deleteAfterDate,omitempty"`
+
+	// Labels is an array containing key-value pairs that tag and categorize the database user.
+	// Each key and value has a maximum length of 255 characters.
+	Labels []LabelSpec `json:"labels,omitempty"`
+
+	// Roles is an array of this user's roles and the databases / collections on which the roles apply. A role allows
+	// the user to perform particular actions on the specified database.
+	Roles []RoleSpec `json:"roles"`
+
+	// Scopes is an array of clusters and Atlas Data Lakes that this user has access to.
+	Scopes []ScopeSpec `json:"scopes"`
+
+	// PasswordSecret is a reference to the Secret keeping the user password.
+	PasswordSecret *ResourceRef `json:"passwordSecretRef"`
+
+	// Username is a username for authenticating to MongoDB.
+	Username string `json:"username"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Name",type=string,JSONPath=`.spec.name`
+// +kubebuilder:subresource:status
+// +groupName:=atlas.mongodb.com
+
+// AtlasDatabaseUser is the Schema for the Atlas Database User API
+type AtlasDatabaseUser struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AtlasDatabaseUserSpec          `json:"spec,omitempty"`
+	Status status.AtlasDatabaseUserStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// AtlasDatabaseUserList contains a list of AtlasDatabaseUser
+type AtlasDatabaseUserList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AtlasDatabaseUser `json:"items"`
+}
+
+// RoleSpec allows the user to perform particular actions on the specified database.
+// A role on the admin database can include privileges that apply to the other databases as well.
+type RoleSpec struct {
```

Comment:
It looks like all fields in a `Role` are required, so we need to remove `omitempty`

Reply:
good catch!